packages/flutter_tools/lib/src/base/fingerprint.dart

```diff
@@ -134,14 +134,15 @@ class Fingerprint {
   });
 
   @override
-  bool operator==(dynamic other) {
+  bool operator==(Object other) {
     if (identical(other, this))
       return true;
-    if (other.runtimeType != runtimeType)
-      return false;
-    final Fingerprint typedOther = other;
-    return _equalMaps(typedOther._checksums, _checksums)
-        && _equalMaps(typedOther._properties, _properties);
+    if (other is Fingerprint) {
```

Comment:
we should compare runtimeType not use `is`, in an `==` operator; see style guide

Reply:
comparing runtimeType doesn't do a type promotion, so we need to cast regardless. I'd rather have the type safety of the `is` check, and I think we should allow the tool to diverge from the framework in this regard - especially since there are only a handful of types that implement `operator.==` here.

Reply:
the problem is it'll break when you subclass Fingerprint, and it'll break when someone copies and pastes this code into other code where it isn't as safe.

We're going to have to work out how to make this type safe in any case.